airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py

```diff
@@ -132,11 +132,14 @@ def path(self, **kwargs) -> str:
         return "surveys"
 
     def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        survey_ids = params.get("survey_ids", [])
         result = super().parse_response(response=response, stream_state=stream_state, **kwargs)
         for record in result:
-            substream = SurveyDetails(survey_id=record["id"], start_date=self._start_date, authenticator=self.authenticator)
-            child_record = substream.read_records(sync_mode=SyncMode.full_refresh)
-            yield from child_record
+            if not survey_ids or record["id"] in survey_ids:
+                substream = SurveyDetails(survey_id=record["id"], start_date=self._start_date, authenticator=self.authenticator)
+                child_record = substream.read_records(sync_mode=SyncMode.full_refresh)
+                yield from child_record
```

Comment:
This will yield records when the `survey_ids` is empty?

Reply:
i believe so
it starts with 
```
for record in result
```
where `result` returns _all_ surveys

so for each record in result: `if not survey_ids or record["id"] in survey_ids` i.e. if we haven't specified a custom list of surveys, or if the record exists in the custom list of surveys, then yield